src/mlpack/tests/load_save_test.cpp

```diff
@@ -129,6 +180,33 @@ BOOST_AUTO_TEST_CASE(LoadTSVExtensionTest)
   remove("test_file.tsv");
 }
 
+/**
+ * Make sure text in tsv is loaded correctly.
+ */
+BOOST_AUTO_TEST_CASE(LoadTextTSVTest)
+{
+  fstream f;
+  f.open("test_file.tsv", fstream::out);
+
+  f << "1\tfield 2\tfield 3" << endl;
+  f << "2\t\"field 2\t with tab\"\tfield 3" << endl;
+  f << "3\tfield 2 with \"embedded quote\"\tfield 3" << endl;
+  f << "5\t field 2 with embedded \\ \tfield 3" << endl;
+  f << "11\t \tfield 33" << endl;
+
+  f.close();
+
+  arma::mat test;
+  data::DatasetInfo info;
+  BOOST_REQUIRE(data::Load("test_file.tsv", test, info, false, true) == true);
```

Comment:
I think it would be a good idea to compare the actual input/output.

Reply:
Yeah I guess comparing elements would make tests more concrete. I 'll add them.